### PR TITLE
sktime v0.12.1

### PR DESCRIPTION
**Pull request by lmmentel**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Related:
- https://github.com/alan-turing-institute/sktime/issues/2921
- https://github.com/conda-forge/sktime-feedstock/issues/44

### CODE REVIEW

**Comment by lmmentel** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ outputs:
       host:
         - python
         - deprecated >=1.2.13
-        - numba >=0.53,<0.55
+        - numba >=0.53
```
Comment (dhirschfeld) [inline recipe/meta.yaml:32]: This is preventing me from installing `sktime` so good to see this will be fixed in the next release!👍

Comment (lmmentel) [reply recipe/meta.yaml]: no worries, I'll loosen up the restriction, but having some trouble resolving deps for this release.

**Comment by lmmentel** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (lmmentel) [reply recipe/meta.yaml]: `sktime` seems to be working with number version relaxed

**Review by fkiraly — APPROVED** [review]
Exact reason why this works is mysterious to me, approving based on cursory look and assertions of @lmmentel that this will fix the failed release 0.12.1

**Pull request merged**